app/src/main/java/hous/release/android/presentation/todo/daily/DailyFragment.kt

```diff
@@ -0,0 +1,76 @@
+package hous.release.android.presentation.todo.daily
+
+import android.os.Bundle
+import android.view.View
+import androidx.core.content.ContextCompat.getColor
+import androidx.fragment.app.viewModels
+import androidx.lifecycle.flowWithLifecycle
+import androidx.lifecycle.lifecycleScope
+import androidx.navigation.fragment.findNavController
+import dagger.hilt.android.AndroidEntryPoint
+import hous.release.android.R
+import hous.release.android.databinding.FragmentDailyBinding
+import hous.release.android.util.HousFloatingButton
+import hous.release.android.util.binding.BindingFragment
+import kotlinx.coroutines.flow.launchIn
+import kotlinx.coroutines.flow.onEach
+
+@AndroidEntryPoint
+class DailyFragment : BindingFragment<FragmentDailyBinding>(R.layout.fragment_daily) {
+    private val dailyAdapter = DailyAdapter()
+    private val dailyViewModel: DailyViewModel by viewModels()
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        initStatusBarColor()
+        initViewPager()
+        initTabLayout()
+        onClickBackButton()
+        initFloatingButton()
+    }
+
+    private fun onClickBackButton() {
+        binding.ivDailyBackButton.setOnClickListener {
+            findNavController().navigateUp()
+        }
+    }
+
+    private fun initViewPager() {
+        dailyViewModel.dailyToDos
+            .flowWithLifecycle(viewLifecycleOwner.lifecycle)
+            .onEach { dailyTodos -> dailyAdapter.submitList(dailyTodos) }
+            .launchIn(viewLifecycleOwner.lifecycleScope)
+
+        binding.vpDailyTodos.apply {
+            adapter = dailyAdapter
+            isUserInputEnabled = false
+        }
+    }
+
+    private fun initTabLayout() {
+        dailyViewModel.dailyTabCurrIndex
+            .flowWithLifecycle(viewLifecycleOwner.lifecycle)
+            .onEach { currIndex ->
+                binding.cvDailyWeekOfDayTab.setContent {
+                    DailyTab(
+                        currIndex = currIndex,
+                        setCurrIndex = dailyViewModel::setTabCurrIndex
+                    )
+                }
+                binding.vpDailyTodos.currentItem = currIndex
+            }
+            .launchIn(viewLifecycleOwner.lifecycleScope)
+    }
+
+    private fun initFloatingButton() {
+        binding.cvDailyFloatingButton.setContent {
+            HousFloatingButton {
+                /* TO DO 추가하기 뷰로 이동하는 함수 */
+            }
+        }
+    }
+
+    private fun initStatusBarColor() {
+        activity?.window?.statusBarColor = getColor(requireActivity(), R.color.hous_g_1)
+    }
```

Comment:
```suggestion
               safeLet(activity as MainActivity, activity?.window) { activity, window ->
            window.statusBarColor = getColor(activity, R.color.hous_g_1)
        }

inline fun <T1 : Any, T2 : Any, R : Any> safeLet(p1: T1?, p2: T2?, block: (T1, T2) -> R?): R? {
    return if (p1 != null && p2 != null) block(p1, p2) else null
}
```
safeLet 확장함수를 util에 추가해서 위와 같은 방식으로 바꿔보는거 어떠세용??

Reply:
앗.. 이전 하우스 코드 복붙이라 생각치도 못했네요 저도 한번 확장함수 써보겠습니다!!

Reply:
@murjune 그냥 원래 코드 쓸래유... 한.줄.조.와...

Reply:
@murjune 그냥 원래 코드 쓸래유... 한.줄.조.와...